bpf/aya-bpf/src/maps/perf/perf_event_array.rs

```diff
@@ -66,4 +66,12 @@ impl<T> PerfEventArray<T> {
             );
         }
     }
+
+    pub fn skb_output<C: BpfContext>(&self, ctx: &C, skb_len: u32, data: &T) {
+        self.output_at_index(ctx, BPF_F_CURRENT_CPU as u32, data, skb_len)
+    }
```

Comment:
Or actually... to make it more Rusty, what about taking `SkBuff` directly, to not have to provide the length manually?

We could provide a trait which is implemented by `SkBuffContext` and `TcContext`, like:

```rust
pub Trait BpfSkBuffContext {
    fn skb(&self) -> SkBuff;
}
```

(I'm open to suggestious about better naming :stuck_out_tongue: )

And then modify this method to take `BpfSkBuffContext` and take out the burden of providing length:

```suggestion
    pub fn skb_output<C: BpfSkBuffContext>(&self, ctx: &C, data: &T) {
        self.output_at_index(ctx, BPF_F_CURRENT_CPU as u32, data, self.skb.len())
    }
```